examples/config.py

```diff
@@ -0,0 +1,148 @@
+#   distribution.
+# * The name of the author may not be used to endorse or promote
+#   products derived from this software without specific prior
+#   written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""This script allows a user to launch a chain of NFs from a JSON
+config file passed in as an argument when running the script"""
+
+import sys
+import json
+import os
+import shlex
+import time
+from signal import signal, SIGINT
+from subprocess import Popen
+from datetime import datetime
+
+def handler(signal_received, frame):
+    """Handles cleanup on user shutdown"""
+    # pylint: disable=unused-argument
+    for lf in log_files:
+        lf.close()
+    print("\nExiting...")
+    sys.exit(0)
+
+def get_config():
+    """Gets path to JSON config file"""
+    if len(sys.argv) < 2:
+        print("Error: Specify config file")
+        sys.exit(0)
+
+    file_name = sys.argv[1]
+    if os.path.exists(file_name):
+        return file_name
+    print("Error: No config file found")
+    sys.exit(0)
+
+def on_failure(error_output):
+    """Handles shutdown on error"""
+    for lf in log_files:
+        lf.close()
+    for n in nf_list:
+        try:
+            os.system("sudo pkill " + n)
+        except OSError:
+            pass
+    print(error_output)
+    print("Error occurred. Exiting...")
+    sys.exit(1)
+
+def running_services():
+    """Checks running NFs"""
+    while 1:
+        for pl in procs_list:
+            ret_code = pl.poll()
+            if ret_code is not None:
+                out, err = pl.communicate()
+                on_failure(out + err)
+                break
+            time.sleep(.1)
+            continue
+
+procs_list = []
+nf_list = []
+cmds_list = []
+log_files = []
+if __name__ == '__main__':
+    signal(SIGINT, handler)
+    if os.path.basename(os.getcwd()) != "examples":
+        print("Error: Run script from within /examples folder")
+        sys.exit(1)
+
+    if len(sys.argv) < 3:
+        time_obj = datetime.now().time()
+        log_dir = time_obj.strftime("%H:%M:%S")
+    else:
+        log_dir = sys.argv[2]
+    try:
+        os.mkdir(log_dir)
+    except OSError:
+        on_failure("Creation of directory %s failed" % (dir))
```

Comment:
`dir` appears to be a built-in Python function. I think `(dir)` is supposed to be `(log_dir)` here, as you're just calling that method and it's returning standard output from that command:
```
Creation of directory <built-in function dir> failed
Error occurred. Exiting...
```
To verify this theory, open Python 3 and run the `dir` command:
```
$ python3
Python 3.6.9 (default, Apr 18 2020, 01:56:04) 
[GCC 8.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> dir
<built-in function dir>
```

Furthermore, it seems like it's requiring that we allow the script to create a new directory for the log files. While I understand the motivation for this because it'll alleviate the possibility of using a non-empty directory (which could be messy), the error message is a little vague. By saying that it failed to create the directory, it could mean several things, like no permission, a directory already exists, etc.. I tested this by changing the `examples` folder's permissions to prohibit the script from creating the folder and by also trying to run the script in a situation where the directory already exists. Alternatively, we could also allow the script to use a directory that was just created, like I did. Just something to consider, but the main issue here is calling `dir` instead of using `log_dir`. Additionally, if we just wanted to use one directory and run this multiple times, we would be forced to delete the one that already exists and start over.

Reply:
Yep, that was a typo. I meant log_dir, thanks!